src/server/game/Spells/SpellMgr.cpp

```diff
@@ -3261,26 +3261,26 @@ void SpellMgr::LoadSpellCustomAttr()
 
     CreatureAI::FillAISpellInfo();
 
-    sLog->outString(">> Loaded spell custom attributes in %u ms", GetMSTimeDiffToNow(oldMSTime));
+    sLog->outString(">> Loaded SpellInfo custom attributes in %u ms", GetMSTimeDiffToNow(oldMSTime));
     sLog->outString();
 }
 
-inline void ApplySpellFix(std::initializer_list<uint32> spellIds, void(*fix)(SpellEntry*))
+inline void ApplySpellFix(std::initializer_list<uint32> spellIds, void(*fix)(SpellInfo*))
 {
     for (uint32 spellId : spellIds)
     {
-        SpellEntry const* spellInfo = (SpellEntry*)sSpellStore.LookupEntry(spellId);
+        SpellInfo const* spellInfo = (SpellInfo*)sSpellStore.LookupEntry(spellId);
         if (!spellInfo)
         {
             sLog->outErrorDb("Spell info correction specified for non-existing spell %u", spellId);
             continue;
         }
 
-        fix(const_cast<SpellEntry*>(spellInfo));
+        fix(const_cast<SpellInfo*>(spellInfo));
```

Comment:
just curious: why using a `const_cast` instead of `static_cast` here?

Reply:
ask trinity Q.Q

Reply:
````
`const_cast` can be used to remove or add `const` to a variable; no other C++ cast is capable of removing it (not even reinterpret_cast). It is important to note that modifying a formerly `const` value is only undefined if the original variable is const; if you use it to take the `const` off a reference to something that wasn't declared with `const`, it is safe. This can be useful when overloading member functions based on `const`, for instance. It can also be used to add const to an object, such as to call a member function overload.```